File: backend/_tecnicos/Tecnicos/src/main/java/br/ufpr/tcc/MSTecnicos/models/Tecnico.java
package br.ufpr.tcc.MSTecnicos.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.*;

@Entity
public class Tecnico extends Pessoa {

    @Column(name = "matricula")
    @JsonProperty("matricula")
    private String matricula;

    @Column(name = "ativo")
    @JsonProperty("ativo")
    private boolean ativo;

    @Lob
    @Column(name = "foto_perfil")
    @Basic(fetch = FetchType.LAZY)
    private byte[] fotoPerfil;

	public Tecnico() {
		super();
	}
	
	public Tecnico(Long id, String nome, String sexo, int idade, Endereco endereco, String dataNasc, String cpf,
			String telefone, String matricula, boolean ativo) {
		super();
		this.id = id;
		this.nome = nome;
		this.sexo = sexo;
		this.idade = idade;
		this.endereco = endereco;
		this.dataNasc = dataNasc;
		this.cpf = cpf;
		this.telefone = telefone;
		this.matricula = matricula;
		this.ativo = ativo;
	}

    public String getMatricula() {
        return matricula;
    }

    public void setMatricula(String matricula) {
        this.matricula = matricula;
    }

    public boolean isAtivo() {
        return ativo;
    }

    public void setAtivo(boolean ativo) {
        this.ativo = ativo;
    }

    public byte[] getFotoPerfil() {
        return fotoPerfil;
    }

    public void setFotoPerfil(byte[] fotoPerfil) {
        this.fotoPerfil = fotoPerfil;
    }
}

File: frontend/Web/src/app/_shared/models/pessoa/tecnico/tecnico.ts
import { Endereco } from "../endereco";
import { Pessoa } from "../pessoa";

export class Tecnico extends Pessoa {
    constructor(
        public matricula?: number,
        public ativo?: boolean,
        public fotoUrl?: string,
        id?: number,
        nome?: string,
        sexo?: string,
        idade?: number,
        endereco?: Endereco,
        dataNasc?: Date,
        cpf?: string,
        telefone?: string
    ) {
        super(id, nome, sexo, idade, endereco, dataNasc, cpf, telefone);
    }
    static fromJSON(json: any): Tecnico {
        return Object.assign(new Tecnico(), json);
      }
}